tests/Pororoca.Domain.Tests/Features/VariableResolution/IPororocaVariableResolverTests.cs

```diff
@@ -59,6 +59,23 @@ public static void Should_replace_predefined_variable_correctly()
         // THEN
         Assert.Equal($"Today is {todayStr}", resolvedString);
     }
+    
+     [Fact]
+    public static void Should_replace_recursive_variables_correctly()
```

Comment:
Let's add more tests:

- Depth 1, 2, 3 recursion
- Max recursion depth reached (4)
- Recursion with predefined variables
- Recursion when a inner variable is not found